### PR TITLE
isort support

### PR DESCRIPTION
**Pull request by VeckoTheGecko**
[isort](https://pycqa.github.io/isort) sorts your imports so that you don't have to. The tool organises them into [5 sections](https://pycqa.github.io/isort/docs/configuration/custom_sections_and_ordering.html) separated by newlines:

- FUTURE (eg. `from __future__ import annotations`)
- STDLIB (eg. `import os`)
- THIRDPARTY (eg. `import xarray`)
- FIRSTPARTY (eg. `import parcels.Field`)
- LOCALFOLDER (eg. `from .utils import square`)

Within each section the imports are sorted alphabetically, and also squashes imports down to one line.
eg.
```py
from parcels.field import Field
from parcels.field import NestedField
from parcels.field import SummedField
from parcels.field import VectorField
```
becomes
```py
from parcels.field import Field, NestedField, SummedField, VectorField
```

This will make import statements through the codebase much more structured, and easier to read. isort also has good integration with pre-commit.

You can add `# isort:skip` to a line to prevent isort from moving the import statement.

### CODE REVIEW

**Comment by VeckoTheGecko** [comment]
Although this tool does standardize the code (so people don't have to worry about import orders), it does result in a large diff in this PR (and hence soiling git blame).

If having a good git blame is important, might be worth adding a `.git-blame-ignore-revs` file referencing commits that implement large formatting changes so they get [ignored in git blame view](https://github.blog/changelog/2022-03-24-ignore-commits-in-the-blame-view-beta/).

**Comment by VeckoTheGecko** [comment]
That's pretty much it for this feature. Happy to discuss more, or merge if it looks good.

File: parcels/examples/example_stommel.py
Comment (VeckoTheGecko) [inline parcels/examples/example_stommel.py:15]: isort will mess with some of the custom sorting here.

File: tests/test_advection.py
Comment (erikvansebille) [inline tests/test_advection.py:29]: Do you know why there's only one class per line in this import call (and similar files)? From the isort documentation, it seems that multiple classes are also supported. Or is this intentional because it's even cleaner?

**Comment by erikvansebille** [comment]
Thanks for implementing this, @VeckoTheGecko; looks great! I agree with you that using tools like isort to clean up the code (and keep it clean) will facilitate development. So I'm happy to merge into master after you've answered my question above on the one-per-line-import in the test*-files

File: tests/test_advection.py
Comment (VeckoTheGecko) [reply tests/test_advection.py:29]: This is the way imports are formatted using the `black` auto formatter, which `isort` is compatible with by specifying the profile to be `black` (see the modified `setup.cfg`).

Although we don't use `black` at the moment, it is a really popular formatter that I think would be good to consider introducing in future. This is just a small step in that direction, preventing reformatting the imports again should we start using `black`.

**Review by erikvansebille — APPROVED** [review]

**Comment by VeckoTheGecko** [comment]
@erikvansebille would you mind bringing `new_docs` up to date with `master`? I'm wanting to branch off `new_docs` with these new changes :)

**Comment by erikvansebille** [comment]
> @erikvansebille would you mind bringing new_docs up to date with master? I'm wanting to branch off new_docs with these new changes :)

Yup, will do after I also merged #1325. But can't you do it yourself?

**Comment by VeckoTheGecko** [comment]
`new_docs` is on the `parcels` repo, so I have no permissions or write access (outside of submitting PRs into it). So can't sync it with `master`

**Comment by erikvansebille** [comment]
OK, I've just invited you to the OceanParcels team, so that you can commit directly on the OceanParcels repo. Welcome to the team!

**Comment by VeckoTheGecko** [comment]
Glad to be aboard! 🥳🥳🥳🥳

**Pull request merged**
